kolibri/plugins/coach_tools/api.py

```diff
@@ -22,7 +22,18 @@ class KolibriReportPermissions(permissions.BasePermission):
 
     # check if requesting user has permission for collection or user
     def has_permission(self, request, view):
-        return request.user.can_read(get_collection_or_user(view.kwargs))
+        collection_kind = view.kwargs.get('collection_kind', None)
+        collection_id = view.kwargs.get('collection_id', None)
+        user_pk = view.kwargs.get('pk', None)
+
+        if any(collection_kind in kind for kind in collection_kinds.choices):
+            perm_check_obj = Collection.objects.get(pk=collection_id)
+        elif collection_id:
```

Comment:
If you're not explicitly checking that it's "user", would be good to at least as a comment to that effect.